disnake/ext/commands/interaction_bot_base.py

```diff
@@ -1201,26 +1276,42 @@ async def process_application_commands(
         interaction: :class:`disnake.ApplicationCommandInteraction`
             The interaction to process commands for.
         """
-        if self._sync_commands and not self._sync_queued:
+        if self._command_sync._sync_enabled and not self._sync_queued:
             known_command = self.get_global_command(interaction.data.id)  # type: ignore
 
             if known_command is None:
                 known_command = self.get_guild_command(interaction.guild_id, interaction.data.id)  # type: ignore
 
             if known_command is None:
+                # don't do anything if we aren't allowed to disable them
                 # This usually comes from the blind spots of the sync algorithm.
                 # Since not all guild commands are cached, it is possible to experience such issues.
                 # In this case, the blind spot is the interaction guild, let's fix it:
-                try:
-                    await self.bulk_overwrite_guild_commands(interaction.guild_id, [])  # type: ignore
-                except disnake.HTTPException:
-                    pass
+                if self._command_sync.allow_command_deletion:
+                    try:
+                        await self.bulk_overwrite_guild_commands(interaction.guild_id, [])  # type: ignore
+                    except disnake.HTTPException:
+                        message = (
+                            "This command does not exist locally. More information about this: "
```

Comment:
https://github.com/DisnakeDev/disnake/pull/806/commits/04639b9af15aac8a07e4d28b6fc46fadeebbf2dd maybe?

Reply:
Getting a `HTTPException` here doesn't mean the command doesn't exist, errors here would mostly be permission errors (which shouldn't happen as it would never get to this point in the first place).

Reply:
Ah, fair